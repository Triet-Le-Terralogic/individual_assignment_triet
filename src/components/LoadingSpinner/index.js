import React from "react";

export default function LoadingSpinner() {
  return (
    <div className="Loading-spinner" data-test="spinner">
      <div className="Loading-spinner__item" data-test="spinner-item">
        <div></div>
        <div></div>
      </div>
    </div>
  );
}
