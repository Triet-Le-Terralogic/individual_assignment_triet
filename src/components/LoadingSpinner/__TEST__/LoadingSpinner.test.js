import React from "react";
import LoadingSpinner from "../index";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<LoadingSpinner />);
let container, containerProps, loadingSpinnerItem;

describe("<LoadingSpinner/>", () => {
  beforeEach(() => {
    container = wrapper.find("[data-test='spinner']");
    containerProps = container.props();
    loadingSpinnerItem = container.find("[data-test='spinner-item']");
  });

  it("should render withour crash", () => {
    shallow(<LoadingSpinner />);
  });

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render a div.Loading-spinner container", () => {
    expect(container).toHaveLength(1);
    expect(containerProps.className).toContain("Loading-spinner");
  });

  it("should render a div.Loading-spinner__item inside container", () => {
    expect(loadingSpinnerItem).toHaveLength(1);
    expect(loadingSpinnerItem.props().className).toContain(
      "Loading-spinner__item"
    );
  });

  it("should contain 2 empty div", () => {
    expect(loadingSpinnerItem.children()).toHaveLength(2);
    expect(loadingSpinnerItem.childAt(0).text()).toMatch("");
    expect(loadingSpinnerItem.childAt(1).text()).toMatch("");
  });
});
