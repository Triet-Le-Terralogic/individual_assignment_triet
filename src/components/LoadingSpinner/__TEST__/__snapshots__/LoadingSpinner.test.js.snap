// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner/> should render correctly 1`] = `
<div
  className="Loading-spinner"
  data-test="spinner"
>
  <div
    className="Loading-spinner__item"
    data-test="spinner-item"
  >
    <div />
    <div />
  </div>
</div>
`;
